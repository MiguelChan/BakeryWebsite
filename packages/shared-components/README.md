# `@mgl/shared-components`

> TODO: description

## Usage

```
const sharedComponents = require('@mgl/shared-components');

// TODO: DEMONSTRATE API
```
