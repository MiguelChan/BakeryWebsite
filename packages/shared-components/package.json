{
  "name": "@mgl/shared-components",
  "version": "1.1.0",
  "description": "The WebApp components library",
  "main": "./build/src/index.js",
  "types": "./build/src/index.d.ts",
  "files": [
    "./build/src/index.js"
  ],
  "scripts": {
    "clean": "rm -rf build",
    "build": "tsc",
    "test": "jest --config jest.config.js",
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint . --ext .ts --fix",
    "release": "npm run clean && npm run lint && npm run test && npm run build"
  },
  "devDependencies": {
    "eslint-plugin-modules-newline": "^0.0.6"
  },
  "peerDependencies": {}
}
