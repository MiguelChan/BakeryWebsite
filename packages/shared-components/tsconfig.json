{
    "compilerOptions": {
      "target": "ESNext",
      "module": "commonjs",
      "outDir": "./build",
      "declarationDir": "build",
      "declaration": true,
      "lib": ["ESNext"],
      "types": ["node", "jest"]
    },
    "include": [
      "src/**/*",
      "tst/**/*"
    ],
    "exclude": [
        "node_modules",
        "build"
    ]
  }