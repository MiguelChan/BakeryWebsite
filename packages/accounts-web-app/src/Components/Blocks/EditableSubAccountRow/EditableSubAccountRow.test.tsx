import { SubAccount } from '@mgl/shared-components';
import {
  Table,
  TableBody,
} from '@mui/material';
import {
  RenderResult,
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import React from 'react';
import {
  DELETE_BUTTON,
  EditableSubAccountRow,
  EditableSubAccountRowProps,
  TEXT_FIELD,
} from './EditableSubAccountRow';
import {
  Primary,
} from './EditableSubAccountRow.stories';

describe('EditableSubAccountRow', () => {
  const setupComponent = (props: EditableSubAccountRowProps): RenderResult => {
    const Component = (
      <Table>
        <TableBody>
          <EditableSubAccountRow {...props} />
        </TableBody>
      </Table>
    );

    return render(Component);
  };

  const initialSubAccount = (): SubAccount => ({
    description: 'ADescription',
    id: 'SomeId',
  });

  it('Should display the EditableSubAccountRow', () => {
    const renderResult = setupComponent(Primary.args!);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should call the OnDeleteClickListener when the Delete Icon is clicked', () => {
    const expectedSubAccount = initialSubAccount();
    const mockOnDelete = jest.fn();

    const props: EditableSubAccountRowProps = {
      onDeleteSubAccountClickListener: mockOnDelete,
      onSubAccountUpdatedListener: jest.fn(),
      subAccount: expectedSubAccount,
      readOnly: false,
    };

    setupComponent(props);

    expect(screen.getByTestId(DELETE_BUTTON)).toBeInTheDocument();

    fireEvent.click(screen.getByTestId(DELETE_BUTTON));

    expect(mockOnDelete).toHaveBeenCalledWith(expectedSubAccount);
  });

  it('Should not display the DeleteButton when the Account cant be editted', () => {
    const expectedSubAccount = initialSubAccount();
    expectedSubAccount.id = 'sbacct12345678912341';
    const expectedText = 'SomeText';

    const mockOnUpdate = jest.fn();

    const props: EditableSubAccountRowProps = {
      onDeleteSubAccountClickListener: jest.fn(),
      onSubAccountUpdatedListener: mockOnUpdate,
      subAccount: expectedSubAccount,
      readOnly: false,
    };

    setupComponent(props);

    expect(screen.queryByTestId(DELETE_BUTTON)).toBeNull();

    fireEvent.change(screen.getByTestId(TEXT_FIELD), {
      target: {
        value: expectedText,
      },
    });

    expect(mockOnUpdate).toHaveBeenCalledWith({
      ...expectedSubAccount,
      description: expectedText,
    });
  });

  it('Should call the onSubAccountUpdatedListener when the Name changes', () => {
    const expectedSubAccount = initialSubAccount();
    const expectedText = 'This is the new Text';
    const mockOnUpdate = jest.fn();

    const props: EditableSubAccountRowProps = {
      onDeleteSubAccountClickListener: jest.fn(),
      onSubAccountUpdatedListener: mockOnUpdate,
      subAccount: expectedSubAccount,
      readOnly: false,
    };

    setupComponent(props);

    expect(screen.getByTestId(TEXT_FIELD)).toBeInTheDocument();

    fireEvent.change(screen.getByTestId(TEXT_FIELD), {
      target: {
        value: expectedText,
      },
    });

    expect(mockOnUpdate).toHaveBeenCalledWith({
      ...expectedSubAccount,
      description: expectedText,
    });
  });
});
