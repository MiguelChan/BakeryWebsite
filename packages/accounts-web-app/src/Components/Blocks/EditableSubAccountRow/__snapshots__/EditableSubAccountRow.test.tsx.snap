// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableSubAccountRow Should display the EditableSubAccountRow 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        />
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="description"
              id="description-label"
            >
              Descripcion Subcuenta
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="ATextField"
                id="description"
                name="description"
                type="text"
                value="A description"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Descripcion Subcuenta
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
        >
          <span
            aria-label="delete-subaccount"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
            data-testid="DeleteButton"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
